specs/altair/validator.md

```diff
@@ -0,0 +1,367 @@
+
+Each slot some sync committee members are selected to aggregate the `SyncCommitteeSignature`s into a `SyncCommitteeContribution` which is broadcast on a global channel for inclusion into the next block.
+
+##### Aggregation selection
+
+A validator is selected to aggregate based on the computation in `is_sync_committee_aggregator` where `state` is a `BeaconState` as supplied to `get_sync_committee_slot_signature` and `signature` is the BLS signature returned by `get_sync_committee_slot_signature`. 
+The signature function takes a `BeaconState` with the relevant sync committees for the queried `slot` (i.e. `state.slot` is within the span covered by the current or next sync committee period), and the `privkey` is the BLS private key associated with the validator.
+
+```python
+def get_sync_committee_slot_signature(state: BeaconState, slot: Slot, privkey: int) -> BLSSignature:
+    domain = get_domain(state, DOMAIN_SYNC_COMMITTEE_SELECTION_PROOF, compute_epoch_at_slot(slot))
+    signing_root = compute_signing_root(slot, domain)
+    return bls.Sign(privkey, signing_root)
+```
+
+```python
+def is_sync_committee_aggregator(state: BeaconState, signature: BLSSignature) -> bool:
+    modulo = max(1, SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT // TARGET_AGGREGATORS_PER_SYNC_SUBCOMMITTEE)
+    return bytes_to_uint64(hash(signature)[0:8]) % modulo == 0
+```
+
+*NOTE*: the set of aggregators generally changes every slot; however, the assignments can be computed ahead of time as soon as the committee is known
+
+##### Construct sync committee contribution
+
+If a validator is selected to aggregate the `SyncCommitteeSignature`s produced on a subnet during a given `slot`, they construct an aggregated `SyncCommitteeContribution`. 
+
+Given all of the (valid) collected `sync_committee_signatures` from the `sync_committee_{subnet_id}` gossip during the selected `slot` with an equivalent `beacon_block_root` to that of the aggregator, the aggregator creates a `contribution: SyncCommitteeContribution` with the following fields:
+
+###### Slot
+
+Set `contribution.slot = state.slot` where `state` is the `BeaconState` for the slot in question.
+
+###### Aggregate bits
+
+Let `contribution.aggregate_bits` be a `Bitvector[SYNC_COMMITTEE_SIZE]`, where the `index`th bit is set in the `Bitvector` for each corresponding validator included in this aggregate.
+Note that an aggregator needs to find the index in the sync committee (as returned by `get_sync_committee_indices`) for a given validator referenced by `sync_committee_signature.validator_index` and _this_ is the index to set in the `Bitvector`. 
+For example, validator with index `2044` could be at index `3` in the current sync committee so position `3` in the `Bitvector` is set if a signature from this validator is included in the aggregation. 
+Also note that a validator could be included multiple times in a given sync committee such that multiple bits are set for a single `SyncCommitteeSignature`.
```

Comment:
Ah, I see what you are doing here... The aggregator can include for positions outside of the sub committee.

I *think* that subcommittee/aggregate validations will be simpler if we restrict and have validators double broadcast to different subnets.

In the previous aggregation scheme (no subcommittee aggregates), double broadcasting didn't make much sense to me but in the context of multiple sub committees, I think it does.

Note, if you are duplicated within a single subcommittee, aggregator should include both

Reply:
resolved in newer scheme